scm-plugins/scm-git-plugin/src/main/java/sonia/scm/repository/spi/GitHookChangesetProvider.java

```diff
@@ -59,7 +53,8 @@ public GitHookChangesetProvider(GitChangesetConverterFactory converterFactory, R
   public synchronized HookChangesetResponse handleRequest(HookChangesetRequest request) {
     if (response == null) {
       GitHookChangesetCollector collector = new GitHookChangesetCollector(converterFactory, receivePack, receiveCommands);
-      response = new HookChangesetResponse(collector.collectChangesets());
+      collector.collectChangesets();
```

Comment:
I do not particularly like this "side-effect only" call, but I presume this is legacy code copied from somewhere ? The GitHookChangesetCollector looks like it was copied from somewhere.